Source/WTF/wtf/StdLibExtras.h

```diff
@@ -671,6 +671,51 @@ template<typename OptionalType> auto valueOrDefault(OptionalType&& optionalValue
     return optionalValue ? *std::forward<OptionalType>(optionalValue) : std::remove_reference_t<decltype(*optionalValue)> { };
 }
 
+template<typename T, typename U, std::size_t Extent>
+std::span<T, Extent> spanReinterpretCast(std::span<U, Extent> span)
+{
+    RELEASE_ASSERT(!(span.size_bytes() % sizeof(T)));
+    static_assert(std::is_const_v<T> || (!std::is_const_v<T> && !std::is_const_v<U>), "spanCast will not remove constness from source");
+    return std::span<T, Extent> { reinterpret_cast<T*>(const_cast<std::remove_const_t<U>*>(span.data())), span.size_bytes() / sizeof(T) };
+}
+
+template<typename T, std::size_t Extent>
+std::span<const uint8_t, Extent == std::dynamic_extent ? std::dynamic_extent: Extent * sizeof(T)> asBytes(std::span<T, Extent> span)
+{
+    return std::span<const uint8_t, Extent == std::dynamic_extent ? std::dynamic_extent: Extent * sizeof(T)> { reinterpret_cast<const uint8_t*>(span.data()), span.size_bytes() };
+}
+
+template<typename T, std::size_t Extent>
+std::span<uint8_t, Extent == std::dynamic_extent ? std::dynamic_extent: Extent * sizeof(T)> asWritableBytes(std::span<T, Extent> span)
+{
+    return std::span<uint8_t, Extent == std::dynamic_extent ? std::dynamic_extent: Extent * sizeof(T)> { reinterpret_cast<uint8_t*>(span.data()), span.size_bytes() };
+}
+
+template<typename T, std::size_t Extent1, std::size_t Extent2>
+bool equalSpans(std::span<T, Extent1> a, std::span<T, Extent2> b)
+{
+    if (a.size() != b.size())
```

Comment:
You could make an immediate path (or potentially just disallow with a static_assert) for when Extent1 != Extent2.

```c++
if constexpr (Extent1 != Extent2) {
    return false
} else {
    if (a.size() != b.size())
        return false;
    return !memcmp(a.data(), b.data(), a.size() * sizeof(T));
}
```

Reply:
This doesn't seem right? What if `Extent1` is `std::dynamic_extent` and `Extent2` is 4, the spans could still be equal.

Reply:
Also, I think is currently very efficient for spans with non-dynamic extent given the implementation of `size()` for such types:
```
constexpr size_type size() const noexcept { return _Extent; }
```